.github/workflows/stalebot.yml

```diff
@@ -50,3 +53,88 @@ jobs:
                   enable-statistics: true
                   ascending: true
                   operations-per-run: 120
+            - name: Remove stale PR approvals (over 30 days old)
+              uses: actions/github-script@v7
+              with:
+                  script: |
+                      const thirtyDaysAgo = new Date();
+                      thirtyDaysAgo.setDate(thirtyDaysAgo.getDate() - 30);
+                      console.log(`Checking for approvals before: ${thirtyDaysAgo.toISOString()}`);
+
+                      const prs = await github.paginate(github.rest.pulls.list, {
+                        owner: context.repo.owner,
+                        repo: context.repo.repo,
+                        state: 'open',
+                        per_page: 100
+                      });
+
+                      for (const pr of prs) {
+                        const reviews = await github.paginate(github.rest.pulls.listReviews, {
+                          owner: context.repo.owner,
+                          repo: context.repo.repo,
+                          pull_number: pr.number
+                        });
+                        
+                        // Find the latest approval date for this PR
+                        let latestApproval = null;
+                        let latestApprovalReview = null;
+                        
+                        for (const review of reviews) {
+                          if (review.state === 'APPROVED') {
+                            const reviewDate = new Date(review.submitted_at);
+                            if (!latestApproval || reviewDate > latestApproval) {
+                              latestApproval = reviewDate;
+                              latestApprovalReview = review;
+                            }
+                          }
+                        }
+                        
+                        // Process only if we found an approval
+                        if (latestApproval) {
+                          console.log(`PR #${pr.number} latest approval from ${latestApprovalReview.user.login} on ${latestApproval.toISOString()}`);
+                          
+                          if (latestApproval < thirtyDaysAgo) {
+                            console.log(`Dismissing stale approval on PR #${pr.number} from ${latestApprovalReview.user.login} (approved on ${latestApproval.toISOString()})`);
+                            await github.rest.pulls.dismissReview({
```

Comment:
Looks like we'll dismiss only the latest approval here. WDYTH about dismissing all approvals similar to `remove-draft-approvals`?

Reply:
That was something I hadn't considered, thanks for bringing that up @kalessil.  The latest changes should dismiss all approvals on a PR if the latest approval is more than 30 days old.